libiop/protocols/encoded/lincheck/basic_lincheck_aux.tcc

```diff
@@ -92,9 +93,24 @@ std::shared_ptr<std::vector<FieldT>> multi_lincheck_virtual_oracle<FieldT>::eval
     /* p_{alpha}^1 in [BCRSVW18], but now using the lagrange polynomial from 
      * [TODO: cite Succinct Aurora] instead of powers of alpha. */
     /* Compute p_alpha_prime. */
-    std::vector<FieldT> p_alpha_prime_over_codeword_domain = 
+    std::vector<FieldT> p_alpha_prime_over_codeword_domain;
+    std::vector<FieldT> vd_vp_evaluations;
+    std::vector<FieldT> cd_vp_evaluations;
+    if (this->variable_domain_.num_elements() > this->constraint_domain_.num_elements()){
```

Comment:
Can you swap this if / else? Also can you move the `vd_vp_evaluations` and `cd_vp_evaluations` variable declarations into the if statement scope?